shop/src/test/java/ro/msg/learning/shop/unit/LocationStrategyTest.java

```diff
@@ -0,0 +1,184 @@
+package ro.msg.learning.shop.unit;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import ro.msg.learning.shop.domain.*;
+import ro.msg.learning.shop.domain.key.OrderDetailKey;
+import ro.msg.learning.shop.domain.key.StockKey;
+import ro.msg.learning.shop.service.LocationService;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.service.StockService;
+import ro.msg.learning.shop.service.strategy.MostAbundantStrategy;
+import ro.msg.learning.shop.service.strategy.SingleLocationStrategy;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class LocationStrategyTest {
+    @Mock
+    private LocationService locationService;
+
+    @Mock
+    private ProductService productService;
+
+    @Mock
+    private StockService stockService;
+
+    @InjectMocks
+    private SingleLocationStrategy singleLocationStrategy;
+
+    @InjectMocks
+    private MostAbundantStrategy mostAbundantStrategy;
+
+
+    @Test
+    void testSingleLocationStrategy() {
+
+        StockService stockService = mock(StockService.class);
+        ProductService productService = mock(ProductService.class);
+
+        singleLocationStrategy = new SingleLocationStrategy(stockService, productService);
+
+        Address address1 = new Address("a11", "a12", "a13", "a14");
+        Location location1 = new Location("l1", address1);
+        location1.setId(UUID.randomUUID());
+        Location location2 = new Location("l2", address1);
+        location2.setId(UUID.randomUUID());
+
+        ProductCategory productCategory1 = new ProductCategory("basic", "d1");
```

Comment:
You have duplicated lines in these methods. Use `@BeforeAll` of `@BeforeEach` for the common code.